lib/core/MantraModel.js

```diff
@@ -0,0 +1,97 @@
+import get from 'lodash/get';
+import has from 'lodash/has';
+import set from 'lodash/set';
+import cloneDeep from 'lodash/cloneDeep';
+
+import { error } from './../utils';
+
+export default class MantraModel {
+    constructor({ name, schema }) {
+        this._isConstructorValid({ name, schema });
+        this._path = 'model';
+        const CONFIG = this._getConfig({ name, schema });
+        this.setName(CONFIG);
+        this.setFields(CONFIG, schema);
+    }
+
+    /**
+     * Set the name property
+     * @param {Object} - MantraModel configuration Object
+     */
+    setName(config) {
+        this.name = this.getName(config);
+    }
+
+    /**
+     * Resolve name property
+     * @param {Object} - MantraModel configuration Object
+     * @returns {string || undefined} - Value to be set on the name property.
+     */
+    getName(config) {
+        if (config.name) return config.name;
+        return;
+    }
+
+    /**
+     * Resolve the MantraModelConfig needed from the schema
+     * @param {Object} { name, schema } - The constructor params
```

Comment:
I think, there should be two parameters reflected in the JSDoc because the function `setFields` actually receives two parameters: `config` which is an object and `schema` which is also an object.

My suggestion is the following:
```suggestion
     * @param {Object} config - [Here goes what it's used for]
     * @param {Object} schema - [Here goes what it's used for]
```